File: javascript/pages/MyAccountPage.js
const { expect } = require('@playwright/test');
const { BasePage } = require('../pages/BasePage.js');

exports.MyAccountPage = class MyAccountPage extends BasePage{

  /**
   * @param {import('@playwright/test').Page} page
   */
  constructor(page) {
    super(page);
    this.myAccountH1 = this.page.locator('h1');
  }
  
  async validateMyAccountText(text){
    await expect(this.myAccountH1).toHaveText(text);
  }
  
}

File: java/src/test/java/pages/MyAccountPage.java
package pages;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;
import io.qameta.allure.Step;
import pages.components.MenuComponent;

public class MyAccountPage extends BasePage {
    private final Locator myPersonalInformationButton;

    public MenuComponent menuComponent;

    public MyAccountPage(Page page) {
        super(page);
        menuComponent = new MenuComponent(page);
        myPersonalInformationButton = page.locator(".icon-user");
    }

    @Step("Assert that my personal information button is visible")
    public void assertThatMyPersonalInformationButtonIsVisible() {
        assertIfLocatorIsVisible(myPersonalInformationButton);
    }
}
